### ISSUE TITLE
upgrade to TF .12

### ISSUE THREAD

**Issue by on4tux** [prudhvitella/terraform-provider-infoblox#49]
according to [https://www.terraform.io/upgrade-guides/0-12.html#upgrading-terraform-providers](https://www.terraform.io/upgrade-guides/0-12.html#upgrading-terraform-providers), .11 providers need to upgraded to work with TF .12

**Comment by GraemeKnights** [comment]
I second this as being very desirable if somebody can manage it please? :)
No idea how much work is involved...

**Comment by RobCannon** [comment]
I have converted this to support terraform v0.12.  I have one pull request that needs to be merged first, and then a second PR that builds on top of that.

**Comment by RobCannon** [comment]
Plus, I have support for using the new VS Code container build technology.  You don't have to worry about what version of Go you have installed locally (or you don't even need it installed!).  The container defines the development environment.  It works pretty well.  So, two more PR's waiting once #56 is accepted.

**Comment by JackBracken** [comment]
Thanks @RobCannon 
Give me a few hours while I catch up to the other issues and PRs and I'll test. :)

**Comment by surskitt** [comment]
Apologies for asking both on here and the pull request, but is this still being considered? Would be really nice to have the provider working with terraform 0.12

**Comment by codyja** [comment]
Looking forward to this PR also! Thanks

**Comment by zackhorvath** [comment]
Currently in this boat as well - love the plugin!

**Comment by pescetto** [comment]
@RobCannon @prudhvitella You know if that PR is still open? I don't see it in there and I didn't see it being merged. Just hoping to get this provider updated soon. Thanks to all the contributors who've worked on this!

**Comment by RobCannon** [comment]
#56 is still open.  I have moved onto another company, so i can't really test this module any more.  I have the checkins and can submit them as PRs once #56 is committed, but I can't really support them.

**Comment by pescetto** [comment]
@prudhvitella any time frame on this if possible?

**Comment by surskitt** [comment]
@RobCannon you mentioned you can't support, would this mean you're not willing and/or able to create a fork and merge both of your pull requests into it?

**Comment by RobCannon** [comment]
My code is all here:
https://github.com/RobCannon/terraform-provider-infoblox

There are three branches, one supports modules, one is the update for terraform 0.12 and the third branch adds support for using vscode remote docker support (very handy to make sure all the right requirements are there for building).

**Comment by surskitt** [comment]
Thanks a lot @RobCannon . So do I just need to merge them on top of each other?

**Comment by RobCannon** [comment]
They are all branched from each other.  So the vscoderemote has all of the changes.

**Comment by surskitt** [comment]
@RobCannon I've built from your branches and it all works great! I feel bad for taking so long to thank you.

All the more reason to get these PRs merged.

**Comment by ooOOJavaOOoo** [comment]
@prudhvitella any feedback on merging these?

**Comment by ooOOJavaOOoo** [comment]
@RobCannon i tried building from your branches and I'm still running into the license acceptance error.  Possible I am building something incorrectly?

@prudhvitella are you still supporting this project, i haven't seen any responses from you on any threads

**Comment by RobCannon** [comment]
@ooOOJavaOOoo Sorry, I am not familiar "license acceptance error".

**Comment by ooOOJavaOOoo** [comment]
@RobCannon sorry, had too many github threads going, getting the nul block error

Error: Provider produced invalid plan

Provider "infoblox" planned an invalid value for
module.infoblox_linux.infoblox_record_host.general[0].ipv6addr: attribute
representing a list of nested blocks must be empty to indicate no blocks, not
null.

Using terraform 0.12.8

**Comment by RobCannon** [comment]
Sorry, that it why I can't really support this.  I don't have access to an F5 to try to replicate the issue.  

Plus, I don't think that issue is a result of my changes.  I didn't really make any changes to the core code.  It might be due to a change in terraform types, through.

**Comment by wenesak** [comment]
How do you compile this provider?
